pkg/kubelet/apis/config/types.go

```diff
@@ -100,6 +100,8 @@ type KubeletConfiguration struct {
 	// volumePluginDir is the full path of the directory in which to search
 	// for additional third party volume plugins.
 	VolumePluginDir string
+	// cloudProvider is the provider for cloud services.
+	CloudProvider string
```

Comment:
is there any validation needed in ValidateKubeletConfiguration?

Reply:
the validation is made on InitCloudProvider, not in the flag, should it be moved to the ValidateKubeletConfiguration?

Reply:
anything that can be changed via the config file should be validated as much as possible in ValidateKubeletConfiguration... that is what protects us from picking up a bad config value in an dynamic config update. cc @mtaufen

Reply:
/cc @cheftako 
for opinion on what we should validate up-front and what should be left to the cloud-provider integration

Reply:
My primary opinion is that we should deprecate this. The goal is to get rid of it in a year (From Kubelet, Kube API Server and Kube Controller Manager).

Reply:
Also I believe the format of the config file is dependent upon the cloud provider implementation. So we could see what it would take to call add and call a validate method from kubelet's validate. Given that its deprecated, that may be overkill.